### ISSUE TITLE
diff between two relative path files still shows up as deleted+added

### ISSUE THREAD

**Issue by tstromberg** [chainguard-dev/malcontent#599]
I think we still have our logic a bit off if you are passing two filenames as arguments:

```
go run ./cmd/mal diff ../malcontent-samples/linux/clean/ls.x86_64 ../malcontent-samples/macOS/clean/ls
├─ 🟡 Deleted: ls.x86_64 [MEDIUM]
│     ≡ data [LOW]
│       🟢 compression/lzma — works with lzma files
│     ≡ discovery [LOW]
│       🟢 system/hostname_get — get computer host name: gethostname
│     ≡ execution [LOW]
│       🟢 shell/TERM — Look up or override terminal settings: TERM
│     ≡ filesystem [LOW]
│       🟢 link_read — read value of a symbolic link: readlink
│     ≡ networking [LOW]
│       🟢 url/embedded — contains embedded HTTPS URLs:
│            https://gnu.org/licenses/gpl.html, https://translationproject.org/team/, https://wiki.xiph.org/MIME_Types_and_Fil…
│     ≡ process [MEDIUM]
│       🟡 name_set — get or set the current process name: __progname
│
├─ 🟢 Added: ls [LOW]
│     ≡ execution [LOW]
│       🟢 shell/TERM — Look up or override terminal settings: TERM
│     ≡ filesystem [LOW]
│       🟢 directory/traverse — traverse filesystem hierarchy: _fts_children, _fts_close, _fts_open, _fts_read, _fts_set
│       🟢 link_read — read value of a symbolic link: readlink
```

similar to the `diff(1)` command, this should show up as a single changed file.

**Comment by egibs** [comment]
I think this comes down to the added/removed logic. The first file will be stored in the map with the key `ls.x86_64` and we'll use that to do the lookup for `ls` and it won't match so it will be treated as an add/remove rather than a modify.

**Comment by egibs** [comment]
Ah, I see why. This is from the clean path changes in #617. I'll work on handling this differently for diffs.

**Comment by egibs** [comment]
If I store the value of `rel` without the extension, this works as described above:
```
$ go run cmd/mal/mal.go diff ../malcontent-samples/linux/clean/ls.x86_64 ../malcontent-samples/macOS/clean/ls
├─ 🔵 Changed: ../malcontent-samples/macOS/clean/ls [MEDIUM → LOW]
│     X data [LOW → NONE]
---     🔵 compression/lzma — works with lzma files
│     X discovery [LOW → NONE]
---     🔵 system/hostname — get computer host name
│     ≡ execution [LOW]
│       🔵 shell/TERM — Look up or override terminal settings: TERM
│     ≡ filesystem [LOW]
│       🔵 link_read — read value of a symbolic link: readlink
+++     🔵 directory/traverse — traverse filesystem hierarchy: _fts_children, _fts_close, _fts_open, _fts_read, _fts_set
│     X networking [LOW → NONE]
---     🔵 url/embedded — contains embedded HTTPS URLs
│     X process [MEDIUM → NONE]
---     🟡 name_set — get or set the current process name
│
```

Do we want to go this route? Two files of different types with the same name would show odd behavior drift but maybe we want that? Conversely, two files with different names may be the same underlying file and the diff would show as deleted/added when it would be more appropriate to show as changed.

**Comment by tstromberg** [comment]
FWIW, like diff(1), the file name shouldn't matter at all if two filenames are being passed into it. Filename logic only happens when directories are passed in.

**Comment by tstromberg** [comment]
I mentioned relative paths in the initial bug report, but this happens for absolute paths too:

`go run ./cmd/mal/ diff ../malcontent-samples/linux/clean/ls.x86_64 /bin/ls`

**Comment by egibs** [comment]
> FWIW, like diff(1), the file name shouldn't matter at all if two filenames are being passed into it. Filename logic only happens when directories are passed in.

Ah okay, this clarifies things. I'll hack on that tomorrow.

**Comment by egibs** [comment]
Addressed via #628.
